domains/pallets/executor-registry/src/lib.rs

```diff
@@ -695,8 +708,9 @@ mod pallet {
 
                         total_stake_weight = total_stake_weight.checked_add(&stake_weight).expect(
                             "
-                                `total_stake_weight` as u128 won't overflow even with 100K executor and \
-                                each of them has 1_000_000_000 SSC at stake; qed",
+                            `total_stake_weight` is at least `u128` which ensures \
+                            the overflow never occurs in the real world even with \
+                            100K executor and each of them has 1_000_000_000 SSC at stake; qed",
```

Comment:
I think I commented about this before that it should be tied back to the balance. Then it is easy to argue that no matter how you add stake, total can't physically exceed the size of the balance type by definition.

Reply:
I think we should return as specific arithmetic error to runtime IMO

Reply:
Well, it's tied to the balance at the moment, but not necessarily always the case, for instance, if we introduce some kind of time-weighted stake weight, your stake weight will be 10x stake_amount if you plan to lock at least one year. The stake weight deserves its own type IMHO.

This is inside `on_initialize` so it's not allowed to return an error.

I'd simply change this to an explicit panic if no objection.

```diff
diff --git a/domains/pallets/executor-registry/src/lib.rs b/domains/pallets/executor-registry/src/lib.rs
index 9536dd821..acaf1ce00 100644
--- a/domains/pallets/executor-registry/src/lib.rs
+++ b/domains/pallets/executor-registry/src/lib.rs
@@ -706,12 +706,11 @@ mod pallet {

                         let stake_weight: T::StakeWeight = executor_config.stake.into();

-                        total_stake_weight = total_stake_weight.checked_add(&stake_weight).expect(
-                            "
-                            `total_stake_weight` is at least `u128` which ensures \
-                            the overflow never occurs in the real world even with \
-                            100K executor and each of them has 1_000_000_000 SSC at stake; qed",
-                        );
+                        total_stake_weight = total_stake_weight
+                            .checked_add(&stake_weight)
+                            .unwrap_or_else(|| {
+                                panic!("Domain bundle election can not work properly due to `total_stake_weight` overflow")
+                            });

                         executor_weights.insert(who, stake_weight);
```

Reply:
If this can be flexible then I'm with Ved on returning an error instead. We should only panic when we can guarantee panic will not happen. Looks like we can't in this case.